cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -212,11 +229,15 @@ protected void startUp() throws Exception {
 
   @Override
   protected void shutDown() throws Exception {
+    shutDownUnrequiredServices();
+    datasetService.stopAndWait();
+  }
+
+  private void shutDownUnrequiredServices() {
```

Comment:
we want to call this method on `stopPreview` method, right ? for other places(completed, killed), we can keep them as is i believe.